.github/workflows/test.yml

```diff
@@ -24,6 +24,22 @@ jobs:
           node-version: ${{ matrix.node }}
       - run: npm ci
       - run: npm test
+  typescript_matrix:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        typescript:
+          - 4.1
+          - 4.2
+    name: TypeScript ${{matrix.typescript}}
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v2
+        with:
+          node-version: 14
+      - run: npm ci
+      - run: npm i typescript@${{matrix.typescript}}
```

Comment:
As mentioned in the issue, is the extra time of this PR Check something to consider? How much time it takes with and without this check?

Reply:
The full jest suite takes around `63s` when I ran the action locally, using [act](https://github.com/nektos/act)

Reply:
> The full jest suite takes around `63s` when I ran the action locally, using [act](https://github.com/nektos/act)

GitHub Actions in production will be faster than that right? (unless you have a NASA 💻  @wolfy1339 🤣 ). To me is an _acceptable_ timing and... we can always invest in improving check run times in a different issue if it becomes a problem

Reply:
> GitHub Actions in production will be faster than that right?

They seem to be usually around 30-40s
>(unless you have a NASA computer @wolfy1339 :rofl:)

Nah, nothing of the sorts